### DISCUSSION TITLE
Measurement Values as Strings question of athena concept id  meas value  assignments that would be relevant to lookup in Atlas for Data Challenge

### DISCUSSION THREAD

**Discussion by heschmidt04** [chorus-ai/StandardsModule#99] [Flowsheet Questions] [answered]
Starting a conversation because I don't think we have Measurements as Strings covered in our Delphi Spreadsheet for what becomes their Athena concept ids in the OMOP Measurement table in the value_as_concept_id column. 
I did not know the relationship was there until August 15th 2024 Bridge2AI Office Hours so if this is already handled somewhere, let me know. 

@p-talapova and @jshoughtaling -- I could use some instruction - especially with what we want to focus on for the future Data Challenge. 

Do we only want to define what are an acceptable set of concept ids for specific flowsheets measurement values that are strings? 

We have some flowsheet names that are compacted values. 
I can split out the individual parts of the compacted values separated in this case by ; - semi-colons. 
Separately - I also have delimiters with-in values I've already split on ; - semi-colons which I am not including in this conversation. 

In order to properly track them down in Atlas downstream - 
My question related to what I am seeing - 

**What is a good assignment of an Athena concept id for OMOP Measurement column value_as_concept_id for the measurement values that are strings?** 

We don't have much in the Delphi Spreadsheet for tie in "Meas Value" with class "Answer" or "Qualifier Value" [here](https://docs.google.com/spreadsheets/d/1EH61Y1xuNxei6CT_VcU0AeYY88Gk5aYExaF6THhzQ1U/edit?gid=0#gid=0). 

For example: 

```
Flowsheet Name  - Sample compacted values       
CARDIAC RHYTHM - Normal sinus rhythm;Heart block 
```

For example - Could we assign "Normal sinus rhythm" to ["Sinus Rhythm"](https://athena.ohdsi.org/search-terms/terms/45877096) ?

Would we try to put the "Heart Block" in as ["Heart"](https://athena.ohdsi.org/search-terms/terms/36308337) or as ["Cardiac Rhythm Disturbance"](https://athena.ohdsi.org/search-terms/terms/36307931)? 

For another flowsheet 
```
Flowsheet Name  - Sample values  
PAIN BODY LOCATION 1 - Chest;Incision 
```

Would we be able to use ["Surgical Incision"](https://athena.ohdsi.org/search-terms/terms/45877371) for Incision? 

And for GI tube status would we be able to put ["Feeding tube"](https://athena.ohdsi.org/search-terms/terms/36310833)? or should we define values for Feeding and for Medication Administration, Vented, Water bolus, etc? 

```
Flowsheet Name  - Sample values
GI TUBE STATUS - Feeding;Intermittent medication administration;Vented;Water bolus

```

Thanks!
Heidi

**Comment by heschmidt04** [comment]
Found trach references under these flowsheet names and want to standardize how to break it out and assign a Athena Concept ID in the Meas Value domain for the Measurement value_as_concept_id column. 

Thanks!
Heidi 

```
Flowsheet Name  - Sample values
PHS IP SUCTION TYPE - Oral;Endotracheal tube and Endotracheal tube;Oropharyngeal

Flowsheet Name  - Sample values
HEENT VOICE - No voice;Trach and Speaking valve;Trach and Trach;No voice and Trach;Unable to Assess and Hoarse;Trach

Flowsheet Name  - Sample values
PHS IP SUCTION TYPE - Tracheal;Oral and Oral;Endotracheal tube and a few others

```

Could we use for In Patient Suction Type ["Trach Tube with Suctioning"](https://athena.ohdsi.org/search-terms/terms/45879359) for example?

There isn't a "hoarse" value nor a good match for "speaking valve" 
There is a value for Oropharyngeal under  https://athena.ohdsi.org/search-terms/terms/36307972

So, should we poll the source_values and see when to go meta or when to get specific based on what Atlas would define as a standard concept set?

**Comment by p-talapova** [comment, marked as answer]
Hi Heidi

Indeed, there are no values present in the Delphi data, so we did not perform separate mapping for values. 

> Do we only want to define what are an acceptable set of concept ids for specific flowsheets measurement values that are strings?

In the future, yes, we do plan to define it more precisely. However, at this initial stage, any matching standard concept can be used. Ideally, concepts from the Meas Value domain are preferred, but it's acceptable to use concepts from other domains if necessary for accurate mapping. We hope to analyze the outcomes of value mapping more thoroughly as we progress.

> I can split out the individual parts of the compacted values

You are on the right track by parsing the values. This approach will enable accurate mapping.

> What is a good assignment of an Athena concept id for OMOP Measurement column value_as_concept_id for the measurement values that are strings?

For string measurement values in OMOP, the best approach is to map them to an existing standard concept ID in Athena, preferably from the Meas Value domain, or use a custom 2-billion series concept ID if no direct match exists.

> For example - Could we assign "Normal sinus rhythm" to "Sinus Rhythm" ?

The concept [4276669 "Normal sinus rhythm" (Condition)](https://athena.ohdsi.org/search-terms/terms/4276669) is a better match than simply "Sinus Rhythm." This is because "normal sinus rhythm" specifically refers to the regular, healthy rhythm of the heart, whereas "sinus rhythm" indicates that the electrical impulses are originating from the sinus node, without necessarily implying that the rhythm is normal.

> Would we try to put the "Heart Block" in as "Heart" or as "Cardiac Rhythm Disturbance"?

For Heart Block, there is also a Condition - [320425 Heart block](https://athena.ohdsi.org/search-terms/terms/320425)

> Would we be able to use "Surgical Incision" for Incision?

Yes, we can use this mapping.

> And for GI tube status would we be able to put "Feeding tube"? or should we define values for Feeding and for Medication Administration, Vented, Water bolus, etc?

It would be more accurate to define specific values for each of the different uses, such as "Feeding," "Intermittent medication administration," "Vented," and "Water bolus." Each of these terms represents a distinct status of the GI tube, so grouping them all under a general term like "Feeding tube" might lead to loss of important clinical detail.

> Could we use for In Patient Suction Type ["Trach Tube with Suctioning"](https://athena.ohdsi.org/search-terms/terms/45879359) for example?

I would suggest considering [36310366 Endotracheal tube](https://athena.ohdsi.org/search-terms/terms/36310366) (Meas Value, LOINC) as an option.

> There isn't a "hoarse" value nor a good match for "speaking valve"

If certain terms do not have direct matches in Athena, we recommend making them standard, assigning 2-billion concept_ids, and using **value_as_concept_id**. But it is also possible to keep them as non-standard and populate the **value_source_value** field (though, in this case, these values will not be visible in Atlas).

If possible, could you please collect all unmapped values and share them with the Standards team? We would be happy to analyze these terms and work on further standardization.

> There is a value for Oropharyngeal under https://athena.ohdsi.org/search-terms/terms/3630797

This is a good match.

> So, should we poll the source_values and see when to go meta or when to get specific based on what Atlas would define as a standard concept set?

Yes, we should poll the source values. This will give us a clearer understanding of the data and help us decide whether it's better to map each value to a specific standard concept, use a broader general concept, or standardize any unmapped values to ensure they are visible in Atlas for concept set creation.

**Comment by jshoughtaling** [comment]
@heschmidt04 - excellent question. I believe Polina has answered quite thoroughly above and I agree with all of her responses, but I would just like to add that we are also splitting semi-colon-separated values at Tufts with the following `EXPLODE` logic in Databricks:

```sql
 SELECT
    table_a.*,
    s2c_map.*,
    EXPLODE (SPLIT(table_a.flowsheet_value, ';')) AS meas_value_expanded
FROM
    some_table AS table_a
    INNER JOIN s2c_map AS s2c_map  ON table_a.some_id = s2c_map.some_id

```

Which produces unique rows for each delimited measurement value. For example:

> R BREAKFAST COMPOSITION ->  orange juice; cereal; milk  

Would become an observation with three values:

| DESCRIPTION | CONCEPT_ID | VALUE | VALUE_AS_CONCEPT_ID |
| --- | --- | --- | --- |
| R BREAKFAST COMPOSITION | 4128619 | orange juice | 42904272 |
| R BREAKFAST COMPOSITION | 4128619 | cereal | 4025610|
| R BREAKFAST COMPOSITION | 4128619 | milk | 4323432 |

**Reply by heschmidt04** [reply]
Thanks Jared, I am using similar logic for ; in Snowflake with a inline function called Lateral. It allows for the same splitting of the flowsheet_values into as many records separated by the semi-colon.
